splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/ProjectRestrictOperation.java

```diff
@@ -310,9 +360,14 @@ public boolean apply(ExecRow row) throws StandardException {
         DataSet<ExecRow> sourceSet = source.getDataSet(dsp);
         try {
             operationContext.pushScope();
-			if (restrictionMethodName != null)
-				sourceSet = sourceSet.filter(new ProjectRestrictPredicateFunction<>(operationContext));
-			return sourceSet.map(new ProjectRestrictMapFunction<>(operationContext));
+            // Temporary code to upgrade the source DataSet to a SparkNativeDataSet
+            // to flush out bugs.
+            // TODO:  Remove this after thorough regression testing.
+            if (hasFilterPred() || hasExpressions())
+                sourceSet = sourceSet.upgradeToSparkNativeDataSet(operationContext);
```

Comment:
> I think it's better to remove this conversion, as we keep adding more supported operations we'll get better testing anyway, and this introduces a forced conversion that we are trying to avoid.

Removed the call to upgradeToSparkNativeDataSet.